lib/dialects/mssql/query/compiler.js

```diff
@@ -163,22 +168,36 @@ class QueryCompiler_MSSQL extends QueryCompiler {
     );
   }
 
-  _returning(method, value) {
+  _returning(method, value, returning) {
     switch (method) {
       case 'update':
       case 'insert':
         return value
-          ? `output ${this.formatter.columnizeWithPrefix('inserted.', value)}`
+          ? `output ${this.formatter.columnizeWithPrefix('inserted.', value)} into @out`
           : '';
       case 'del':
         return value
-          ? `output ${this.formatter.columnizeWithPrefix('deleted.', value)}`
+          ? `output ${this.formatter.columnizeWithPrefix('deleted.', value)} into @out`
           : '';
       case 'rowcount':
         return value ? ';select @@rowcount' : '';
     }
   }
 
+  _buildReturningDeclare(values) {
+    if (values) {
```

Comment:
why do additional check if you always check value to be set before using these new functions?

Reply:
If there are no return values I wanted to prevent the use of temp tables. So if no values are being returned then it will not add any unused statements. 

There is an issue here though. Truthy statements do not check length of an array. I will make that fix on both of those statements.